From 883f8ab161758ce5847d43e91a96bf5f2a98a3b1 Mon Sep 17 00:00:00 2001
From: Cosmin-Gabriel Samoila <cosmin.samoila@nxp.com>
Date: Mon, 15 Oct 2018 15:18:09 +0300
Subject: [PATCH 4884/5242] MLK-19936-2 arch: arm64: fsl-imx8mm: add audio
 plls in micfil node

commit  66efd66691cb3be3580199e85d704cb996ce2f0b from
https://source.codeaurora.org/external/imx/linux-imx.git

Parse the clock tree and stop when you find one of the
audio_plls. If this audio_pll cannot support the required
rate, change the parent to the other pll.
Set rate to rate * 1024 so we can support all parameter
configurations with a minimum clock rate.
This is required to support all rates multiple of 11025 and 8000.

Signed-off-by: Cosmin-Gabriel Samoila <cosmin.samoila@nxp.com>
Reviewed-by: Daniel Baluta <daniel.baluta@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi |    8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
index 140bc0c..f3d201e 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
@@ -793,8 +793,12 @@
 			     <GIC_SPI 109 IRQ_TYPE_LEVEL_HIGH>,
 			     <GIC_SPI 110 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&clk IMX8MM_CLK_PDM_IPG>,
-			 <&clk IMX8MM_CLK_PDM_ROOT>;
-		clock-names = "ipg_clk", "ipg_clk_app";
+			 <&clk IMX8MM_CLK_PDM_ROOT>,
+			 <&clk IMX8MM_AUDIO_PLL1_OUT>,
+			 <&clk IMX8MM_AUDIO_PLL2_OUT>,
+			 <&clk IMX8MM_CLK_EXT3>;
+		clock-names = "ipg_clk", "ipg_clk_app",
+			      "pll8k", "pll11k", "clkext3";
 		dmas = <&sdma2 24 26 0x80000000>;
 		dma-names = "rx";
 		status = "disabled";
-- 
1.7.9.5

